x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMHealthBlockedSnapshotIT.java

```diff
@@ -0,0 +1,343 @@
+            GetHealthAction.Request getHealthRequest = new GetHealthAction.Request(true, 1000);
+            GetHealthAction.Response health = admin().cluster().execute(GetHealthAction.INSTANCE, getHealthRequest).get();
+            assertThat(health.getStatus(), equalTo(HealthStatus.YELLOW));
+            HealthIndicatorResult slmIndicator = health.findIndicator(SlmHealthIndicatorService.NAME);
+            assertThat(slmIndicator.status(), equalTo(HealthStatus.YELLOW));
+            assertThat(slmIndicator.impacts().size(), equalTo(1));
+            List<HealthIndicatorImpact> missingSnapshotPolicies = slmIndicator.impacts()
+                .stream()
+                .filter(impact -> SlmHealthIndicatorService.MISSING_SNAPSHOT_IMPACT_ID.equals(impact.id()))
+                .toList();
+            assertThat(missingSnapshotPolicies.size(), equalTo(numUnhealthyPolicy));
+        });
+    }
+
+    private List<String> executePolicies(String node, List<String> policies) throws Exception {
+        List<String> snapshots = new ArrayList<>();
+        for (String policyName : policies) {
+            snapshots.add(executePolicy(node, policyName));
+        }
+        return snapshots;
+    }
+
+    /**
+     * Execute the given policy and return the generated snapshot name
+     */
+    private String executePolicy(String node, String policyId) throws ExecutionException, InterruptedException {
+        ExecuteSnapshotLifecycleAction.Request executeReq = new ExecuteSnapshotLifecycleAction.Request(
+            TEST_REQUEST_TIMEOUT,
+            TEST_REQUEST_TIMEOUT,
+            policyId
+        );
+        ExecuteSnapshotLifecycleAction.Response resp = client(node).execute(ExecuteSnapshotLifecycleAction.INSTANCE, executeReq).get();
+        return resp.getSnapshotName();
+    }
+
+    private void waitForSnapshotsAndClusterState(String repo, List<String> snapshots) throws Exception {
+        for (String snapshot : snapshots) {
+            waitForSnapshot(repo, snapshot);
+        }
+        assertBusy(() -> assertTrue(SnapshotsInProgress.get(internalCluster().clusterService().state()).isEmpty()));
+    }
+
+    private void waitForSnapshot(String repo, String snapshotName) throws Exception {
+        assertBusy(() -> {
+            try {
+                SnapshotsStatusResponse s = getSnapshotStatus(repo, snapshotName);
+                assertThat("expected a snapshot but none were returned", s.getSnapshots().size(), equalTo(1));
+                SnapshotStatus status = s.getSnapshots().get(0);
+                logger.info("--> waiting for snapshot {} to be completed, got: {}", snapshotName, status.getState());
+                assertThat(status.getState(), equalTo(SnapshotsInProgress.State.SUCCESS));
+            } catch (SnapshotMissingException e) {
+                logger.error("expected a snapshot but it was missing", e);
+                fail("expected a snapshot with name " + snapshotName + " but it does not exist");
```

Comment:
I think the log is a bit redundant here, I would suggest doing this instead:
```suggestion
                fail(e, "expected a snapshot with name " + snapshotName + " but it does not exist");
```
I don't think we need this log in every loop that `assertBusy` does. Just including the exception in the `fail` will include it in the `AssertionError` and thus include it in the log at the end of the `assertBusy` (if it fails).

Reply:
removed